File: TranslateRequest.java
package com.resoft.ocr_translation.repository;

import com.resoft.ocr_translation.dto.TranslateRequestDto;
import com.resoft.ocr_translation.dto.TranslateResultRequestDto;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor
public class TranslateRequest {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @JoinColumn(name="user_id")
    private Long userId;

    @JoinColumn(name = "translator_id")
    private Long translatorId;

    @JoinColumn(name = "translate_result_id")
    private Long translate_result_Id;

    @JoinColumn(name = "child_id")
    private Long childId;

    @Column(nullable = false, length = 10)
    private String status;

    public TranslateRequest(TranslateRequestDto translateRequestDto) {
        this.id = translateRequestDto.getId();
        this.userId = translateRequestDto.getUserId();
        this.translatorId = translateRequestDto.getTranslate_result_Id();
        this.childId = translateRequestDto.getChildId();
        this.status = translateRequestDto.getStatus();
    }
}

File: src/main/kotlin/com/example/demo/file/request/entity/TranslateRequest.kt
package com.example.demo.file.request.entity

import com.example.demo.file.auto.entity.AutoTranslate
import com.example.demo.user.basic.type.UserType
import com.example.demo.user.parent.child.entity.Child
import com.example.demo.user.translator.entity.Translator
import jakarta.persistence.*

@Entity
class TranslateRequest(

        @Column(nullable = false, length = 10)
        var status : String,

        @Enumerated(EnumType.STRING)
        var userType : UserType,

        @OneToOne
        @JoinColumn(name = "translator_id")
        var translator : Translator,

        @OneToOne
        @JoinColumn(name = "autoTranslate_id")
        var autoTranslate: AutoTranslate,

        @OneToOne
        @JoinColumn(name = "child_id", nullable = true)
        var child : Child,
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id : Long = -1
}